akka-stream/src/main/scala/akka/stream/IOResult.scala

```diff
@@ -12,7 +16,11 @@ import scala.util.{ Failure, Success, Try }
  * @param count Numeric value depending on context, for example IO operations performed or bytes processed.
  * @param status Status of the result. Can be either [[akka.Done]] or an exception.
  */
-final case class IOResult private[stream] (count: Long, status: Try[Done]) {
+final case class IOResult private (count: Long, status: Try[Done])
+  extends Serializable with Product {
```

Comment:
I'm sorry, but harm was already done since it was a case class from the beginning. It had already `unapply`. We can't fix that after the fact.